libp2p/protocols/pubsub/floodsub.nim

```diff
@@ -32,20 +33,27 @@ const FloodSubCodec* = "/floodsub/1.0.0"
 type
   FloodSub* {.public.} = ref object of PubSub
     floodsub*: PeerTable      # topic to remote peer map
-    seen*: TimedCache[MessageId] # message id:s already seen on the network
-    seenSalt*: seq[byte]
+    seen*: TimedCache[SaltedId] # message id:s already seen on the network
```

Comment:
What about we add the salting to the Data Structures?

Reply:
Fixing `nim` to salt its `Table` is perhaps a fine long-term option but it would take too long for the benefit to show here for it to be practical - it also introduces a tiny bit of overhead so it is not a given that upstream would accept it.

There are some ideas around reimplementing `Table` completely, fixing this and many other things but I don't think there's enough support to warrant the significant effort that would entail (including updating all code to use it)

Reply:
We can create a SecureTable. Would the effort be too big to replace Table where we use salted keys? We could change it incrementally.

Reply:
It seems there should be two implementations or adding a param to the Table constructor instructing it to salt the keys. Salting isn't necessary for all use cases.

Reply:
we could, but it feels out of scope for this PR - ie it already increases safety by using a separate type for the salted id - a rewrite with a special collection is a different task entirely, perhaps best done in isolation